### PR TITLE
Modify version being used to label Sentry releases

### PR DESCRIPTION
**Pull request by iamhitarth**
Made the version used for tracking Sentry releases consistent with the version in Expo's `app.json`.

### CODE REVIEW

**Comment by brentvatne** [comment]
Hi @iamhitarth! The problem here is that `version` isn't necessarily unique every time you publish - we don't enforce that, you can publish with the same `version` field as many times as you like. Instead we assign a unique `revisionId` every time you publish the app, and so we use that here. I hope that makes sense! Maybe instead we should add the `version` somewhere else in the metadata, what do you think?

**Comment by iamhitarth** [comment]
Sorry I am not sure I understand why we need the `revisionId` to be unique? 

If I publish a certain version of my app using Expo, wouldn't we want Sentry to recognize it as that version instead of labelling it as a unique release? 

Feel like I am missing something here in my understanding of the bigger picture.

**Comment by brentvatne** [comment]
Sentry requires that the release is unique: https://github.com/expo/sentry-expo/blob/79c36e9994ee811e3358651365d528118c40aadd/index.js#L42

**Comment by iamhitarth** [comment]
The reasoning behind my changes is that on [Line 23 of index.js](https://github.com/expo/sentry-expo/blob/master/index.js#L23) we try to access a `revisionId` in the manifest - but there isn't one. So I decided to use the `version` instead. 

As a result of that change, the release in the Sentry portal was showing up with the `version` specified in the `app.json`, however, because we are labelling the source maps with the `revisionId` (on [Line 26 of upload-sourcemaps.js](https://github.com/expo/sentry-expo/blob/master/upload-sourcemaps.js#L26) ), the events being tracked with Sentry then started showing up under the release labelled with the `version` instead of the `revisionId`. 

So in conclusion, if we can somehow get the `revisionId` at [Line 23 of index.js](https://github.com/expo/sentry-expo/blob/master/index.js#L23), we would have unique releases with events being logged against the correct version of the sourcemap. 

Have you got any ideas around how we can do that?

**Comment by brentvatne** [comment]
@iamhitarth - if there is no revisionId it means that the project hasn't been published. when you hit publish, each release gets a revisionId. For example, see this app manifest: https://expo.io/@satya164/react-native-paper-example/index.exp?sdkVersion=27.0.0

**Comment by iamhitarth** [comment]
Ah I see. I was convinced that I had published my app and still wasn't seeing the `revisionId` coming through in the app manifest in `index.js`. 

Just checked again though and it seems to be working as expected! 

Thanks a lot for your help @brentvatne ! Will close down this PR now 👍
